contracts/TokenHolderRevenueFund.sol

```diff
@@ -400,57 +453,115 @@ contract TokenHolderRevenueFund is Ownable, AccrualBeneficiary, Servable, Transf
                     controller.getDispatchSignature(), address(this), msg.sender, uint256(amount), currencyCt, currencyId
                 )
             );
-            require(success, "Dispatch by controller failed [TokenHolderRevenueFund.sol:403]");
+            require(success, "Dispatch by controller failed [TokenHolderRevenueFund.sol:456]");
         }
 
         // Emit event
         emit WithdrawEvent(msg.sender, amount, currencyCt, currencyId, standard);
     }
 
+    /// @notice Get the claimable amount for the given wallet-currency pair in the given
+    /// range of accrual period indices
+    /// @param wallet The address of the concerned wallet
+    /// @param currencyCt The address of the concerned currency contract (address(0) == ETH)
+    /// @param currencyId The ID of the concerned currency (0 for ETH and ERC20)
+    /// @param firstPeriodIndex The index of the first accrual period in the range
+    /// @param lastPeriodIndex The index of the last accrual period in the range, clamped to the max period index
+    function claimableAmount(address wallet, address currencyCt, uint256 currencyId,
```

Comment:
Nice! I generally like the idea of query for ranges instead of single index lookup.